crates/vm/levm/src/opcode_handlers/block.rs

```diff
@@ -160,21 +160,27 @@ impl VM {
     ) -> Result<OpcodeSuccess, VMError> {
         self.increase_consumed_gas(current_call_frame, gas_cost::BLOBHASH)?;
 
-        let Ok(index) = TryInto::<usize>::try_into(current_call_frame.stack.pop()?) else {
+        let index = current_call_frame.stack.pop()?;
+
+        let blob_hashes = &self.env.tx_blob_hashes;
+        if index > blob_hashes.len().into() {
             current_call_frame.stack.push(U256::zero())?;
             return Ok(OpcodeSuccess::Continue);
-        };
+        }
 
-        let blob_hashes = &self.env.tx_blob_hashes;
+        let index: usize = index
+            .try_into()
+            .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
 
-        blob_hashes
-            .get(index)
-            .map(|el| {
-                current_call_frame
-                    .stack
-                    .push(U256::from_big_endian(el.as_bytes()))
-            })
-            .unwrap_or_else(|| current_call_frame.stack.push(U256::zero()))?;
```

Comment:
@JereSalo Do you know why we were using map here? I didn't quite understand it.

Were we mapping the Some(value) and doing nothing if it was none?